src/components/Modal/BaseModal.js

```diff
@@ -51,10 +51,9 @@ class BaseModal extends PureComponent {
     }
 
     componentWillUnmount() {
-        // we don't want to call the onModalHide on unmount
-        this.hideModal(this.props.isVisible);
-
+        // Only trigger onClose and setModalVisibility if the modal is unmounting while visible.
         if (this.props.isVisible) {
+            this.hideModal(true);
```

Comment:
There's [concern](https://github.com/Expensify/App/issues/18182#issuecomment-1557827908) about this solution.
cc: @parasharrajat